### DISCUSSION TITLE
Seeking Advice on Automating Blackbox Configuration for Dynamic Endpoint Monitoring with Probes CRD

### DISCUSSION THREAD

**Discussion by Pger-Y** [prometheus-operator/prometheus-operator#7134] [Q&A]
Hi, sorry to bother you. I recently discovered Probes CRD and found it extremely convenient. I'm currently using it as the foundation for my Pingmesh server. However, I've encountered a challenge while attempting to configure Prometheus and Blackbox to monitor my HTTP servers using Probes. My goal is to request numerous endpoints such as http://server1/path with custom-body1 and custom-header1, followed by http://server2/path with custom-body2 and custom-header2, and so on. This list may grow to hundreds of endpoints. It's impractical to keep updating the Blackbox configuration each time new endpoints are added. I am wondering if there are any plans to support this scenario, or if there is something incorrect in my approach. Your guidance would be much appreciated.

**Comment by nicolastakashi** [comment]
Something I have in my bucket list is provide the ability to generate probe resources from a OpenAPI Specification, this might be useful wdyt?

**Reply by Pger-Y** [reply]
I believe that sounds very feasible. However, the key point is that the blackbox-module-config should be dynamically loaded, correct? In the Slack channel, someone mentioned that what we need is a blackbox-operator supporting CRDs similar to Prometheus, which would enable us to manage the blackbox configuration dynamically.

**Reply by nicolastakashi** [reply]
AFAIK we already you have probes CRD you only need configure with a BlackBox Instance, no?
Now if you need manage blackbox, yes we don't support it.

Maybe we should @simonpasquier ?

**Reply by Pger-Y** [reply]
If we need ping probes, the probes CRD works well. However, when monitoring multiple HTTP servers, it's not as convenient.
<img width="531" alt="image" src="https://github.com/user-attachments/assets/3ca814fa-390e-4da1-9dc3-914d7809fe00" />
We might need a blackbox-operator to manage blackbox module configs effectively.

**Comment by nicolastakashi** [comment]
Yes, this is exactly, what I was thinking you wanna a CRD to deploy and Manage BlackBox Exporter.
I think it's a good idea and the first step to achieve that is writing a proposal design doc under: https://github.com/prometheus-operator/prometheus-operator/tree/main/Documentation/proposals @Pger-Y

**Reply by Pger-Y** [reply]
Thank you very much. I will review this and first draft a proposal design. Then, I'll attempt to create a beta version in the office to assess the results.

**Reply by Pger-Y** [reply]
After many days of hard work and long nights, I'm pleased to announce that I have finally completed a draft version of the proposal. Please find it attached in the following link. Your feedback and suggestions would be highly appreciated as we move forward in refining and optimizing this plan. Thank you!

https://github.com/prometheus-operator/prometheus-operator/actions/runs/12908142642/job/35993056610?pr=7294
